moveit_ros/planning_interface/test/subframes_test.cpp

```diff
@@ -199,50 +199,108 @@ TEST(TestPlanUsingSubframes, SubframesTests)
   tf2::Quaternion target_orientation;
   target_orientation.setRPY(0, 180.0 / 180.0 * M_PI, 90.0 / 180.0 * M_PI);
   geometry_msgs::PoseStamped target_pose_stamped;
-  target_pose_stamped.header.frame_id = "box/bottom";
   target_pose_stamped.pose.orientation = tf2::toMsg(target_orientation);
   target_pose_stamped.pose.position.z = Z_OFFSET;
 
   ROS_INFO_STREAM_NAMED(log_name, "Moving to bottom of box with cylinder tip");
+  target_pose_stamped.header.frame_id = "box/bottom";
+  ASSERT_TRUE(moveToCartPose(target_pose_stamped, group, "cylinder/tip"));
```

Comment:
The following block is the same for all subsequent checks. Please factor this out into a function:

```c++
void moveAndValidate(const geometry_msgs::PoseStamped& pose, moveit::planning_interface::MoveGroupInterface& group,
                     const std::string& eef_link, const planning_scene_monitor::PlanningSceneMonitorPtr& psm)
{
  ASSERT_TRUE(moveToCartPose(pose, group, eef_link));
  psm->requestPlanningSceneState();
  planning_scene_monitor::LockedPlanningSceneRO planning_scene(psm);

  // get the tip and box subframe locations in world
  Eigen::Isometry3d eef = planning_scene->getFrameTransform(eef_link);
  Eigen::Isometry3d box_subframe = planning_scene->getFrameTransform(pose.header.frame_id);
  Eigen::Isometry3d target_pose;
  tf::poseMsgToEigen(pose.pose, target_pose);

  // expect that they are identical
  EXPECT_TRUE(eef.isApprox(box_subframe * target_pose, EPSILON)) << "box frame: \n"
                                                                 << box_subframe.matrix() << "\ncylinder frame: \n"
                                                                 << eef.matrix();
}
```

Reply:
I'm not sure though that we should include all these tests at all. They essentially check exactly the same functionality and thus just waste CI time.

Reply:
You can find my suggested changes here: https://github.com/ubi-agni/moveit/tree/p/tylerjw/subframes_test

Reply:
I agree about only checking one of them and that this is just extra redundancy.